sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -386,10 +497,15 @@ public void MarkFailed(Exception? exception)
                     _diagnosticSource?.Write(_activityName + ".Exception", exception);
                 }
 
+#if NETCOREAPP2_1
                 if (ActivityExtensions.SupportsActivitySource())
                 {
                     _currentActivity?.SetErrorStatus(exception?.ToString());
                 }
+#endif
+#if NET6_0_OR_GREATER
```

Comment:
Is there a reason that this block targets >=net6.0 but all the others are simple #else blocks?

Reply:
The SetStatus method is only available in NET 6.0+, this method is used via reflection in the [current implementation](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs#L649), so I think having a no-op for everything NET6.0 and below preserves this behavior